crates/router/src/core/payments/operations/payment_session_intent.rs

```diff
@@ -0,0 +1,306 @@
+{
+    #[instrument(skip_all)]
+    async fn get_customer_details<'a>(
+        &'a self,
+        state: &SessionState,
+        payment_data: &mut payments::PaymentIntentData<F>,
+        merchant_key_store: &domain::MerchantKeyStore,
+        storage_scheme: enums::MerchantStorageScheme,
+    ) -> CustomResult<
+        (
+            BoxedOperation<'a, F, PaymentsSessionRequest, payments::PaymentIntentData<F>>,
+            Option<domain::Customer>,
+        ),
+        errors::StorageError,
+    > {
+        match payment_data.payment_intent.customer_id.clone() {
+            Some(id) => {
+                let customer = state
+                    .store
+                    .find_customer_by_global_id(
+                        &state.into(),
+                        id.get_string_repr(),
+                        &payment_data.payment_intent.merchant_id,
+                        merchant_key_store,
+                        storage_scheme,
+                    )
+                    .await?;
+                Ok((Box::new(self), Some(customer)))
+            }
+            None => Ok((Box::new(self), None)),
+        }
+    }
+
+    #[instrument(skip_all)]
+    async fn make_pm_data<'a>(
+        &'a self,
+        _state: &'a SessionState,
+        _payment_data: &mut payments::PaymentIntentData<F>,
+        _storage_scheme: enums::MerchantStorageScheme,
+        _merchant_key_store: &domain::MerchantKeyStore,
+        _customer: &Option<domain::Customer>,
+        _business_profile: &domain::Profile,
+    ) -> RouterResult<(
+        PaymentsCreateIntentOperation<'a, F>,
+        Option<domain::PaymentMethodData>,
+        Option<String>,
+    )> {
+        Ok((Box::new(self), None, None))
+    }
+
+    async fn get_connector<'a>(
+        &'a self,
+        merchant_account: &domain::MerchantAccount,
+        state: &SessionState,
+        _request: &PaymentsSessionRequest,
+        payment_intent: &storage::PaymentIntent,
+        merchant_key_store: &domain::MerchantKeyStore,
+    ) -> CustomResult<api::ConnectorChoice, errors::ApiErrorResponse> {
+        let db = &state.store;
+        let all_connector_accounts = db
+            .find_merchant_connector_account_by_merchant_id_and_disabled_list(
+                &state.into(),
+                merchant_account.get_id(),
+                false,
+                merchant_key_store,
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Database error when querying for merchant connector accounts")?;
+        let profile_id = &payment_intent.profile_id;
+        let filtered_connector_accounts = helpers::filter_mca_based_on_profile_and_connector_type(
+            all_connector_accounts,
+            profile_id,
+            common_enums::ConnectorType::PaymentProcessor,
+        );
+        let mut connector_and_supporting_payment_method_type = Vec::new();
+        filtered_connector_accounts
```

Comment:
The whole logic on filtered_connector_accounts can be moved into a impl based function. If `Vec<MerchantConnectorAccount>` is not available, create one.